### PR TITLE
Branch bug fix

### LINKED ISSUE #121: [PE-D] Add Person documentation flaw: birthday should be stated in the form of "[]"
Bug Name: birthday should be stated in the form of "[]"

Bug ID: 2

Area Path: User Guide - add

Severity: Low

Description: In your UG, birthday field is stated **optional**, however in your Message usage in the product, it is not described as optional, may cause confusion to users.

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: HangZelin/ped#2

### LINKED ISSUE #130: [PE-D] Suggestion for cosmetic improvement to Help command
Upon starting the app for the first time and typing in "help" in the command box, there is very little allocated space for the text output in the Command Assistant. It's really hard to scroll to read all the text in the tiny window. Perhaps there could be an option to resize the box.

-------------
Labels: `severity.VeryLow` `type.FunctionalityBug`
original: hanif-kamal/ped#1

### LINKED ISSUE #134: [PE-D] Suggestion: add Locale.ENGLISH when displaying birthday
-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: fans2619/ped#4

### LINKED ISSUE #163: [PE-D] Tags can be too long for the display window
Users can create tags that are too long for the GUI window.
The output can be seen below:

Perhaps restrict the number of characters that can be used in a tag.

-------------
Labels: `type.FeatureFlaw` `severity.Medium`
original: MGRL2201/ped#12

### LINKED ISSUE #164: [PE-D] Leap years not accounted for in age calculation in side panel
For a person with birthday on 29 Feb on a leap year (used the year 2000), the side panel indicates that they turn 21 on 29 Feb, but 2021 (current year) isn't a leap year (i.e. 29 Feb 2021 isn't a valid date and shouldn't be reflected). Would be good to account for leap years in the side panel that tracks age of contacts.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: hanif-kamal/ped#8

### LINKED ISSUE #170: [PE-D] No error message when Deletem command is used with incorrect syntax
When inputting Deletem command with incorrect syntax (used "deletem 3-1", where start index > end index), instead of error message being shown for incorrect syntax, the message that appears simply describes the command syntax, even after entering the command. 

As expected, nothing changed in the list. Would be good to add error message for incorrect syntax.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: hanif-kamal/ped#4

### PR DESCRIPTION
**Pull request by Justinhoejj**
Fixes #170 
Fixes #164
Fixes #163 
Fixes #134 
Fixes #130
Fixes #121

### CODE REVIEW

File: src/main/java/seedu/address/commons/core/index/Index.java [hunk 1]
```diff
@@ -9,6 +9,7 @@
  * convert it back to an int if the index will not be passed to a different component again.
  */
 public class Index {
+    public static final String MESSAGE_INVALID_INDEX = "Index is not a non-zero unsigned integer.";
```
Comment (leezhixuan) [inline src/main/java/seedu/address/commons/core/index/Index.java:12]: unsigned?

File: src/main/resources/view/MainWindow.fxml [hunk 2]
```diff
@@ -12,7 +12,7 @@
 <?import javafx.scene.layout.VBox?>
 
 <fx:root type="javafx.stage.Stage" xmlns="http://javafx.com/javafx/8" xmlns:fx="http://javafx.com/fxml/1"
-         title="Address App" minWidth="550" minHeight="600" onCloseRequest="#handleExit">
+         title="Connections" minWidth="1000" minHeight="600" onCloseRequest="#handleExit">
```
Comment (leezhixuan) [inline src/main/resources/view/MainWindow.fxml:15]: All caps instead?

**Review by leezhixuan — APPROVED** [review]
some comments for consideration otherwise LGTM

File: src/main/java/seedu/address/ui/BirthdayReminderCard.java [hunk 3]
```diff
@@ -85,6 +85,21 @@ private String generateBirthdayReminderMessage(Birthday birthday) {
         return String.format(BIRTHDAY_MESSAGE_DEFAULT, age, birthMonthDayAsString);
     }
 
+    private MonthDay accountForLeapDay(MonthDay birthMonthDay) {
+        boolean isLeapDay = birthMonthDay.equals(MonthDay.of(2, 29));
+        boolean hasPassedThisYear = birthMonthDay.isAfter(MonthDay.now());
+        int thisYear = Year.now().getValue();
+        if (!hasPassedThisYear && isLeapDay && !Year.of(thisYear).isLeap()) {
```
Comment (yucheng11122017) [inline src/main/java/seedu/address/ui/BirthdayReminderCard.java:92]: I think you can simply the isLeap() part to just be Year.now().isLeap(), without having to getValue(), year it again and then check if it isLeap :)

**Review by yucheng11122017 — APPROVED** [review]
LGTM except for some minor stuff!

File: src/main/java/seedu/address/ui/BirthdayReminderCard.java (hunk 3 above)
Comment (Justinhoejj) [reply src/main/java/seedu/address/ui/BirthdayReminderCard.java:92]: :)

**Pull request merged**